cmake/CubBuildCompilerTargets.cmake

```diff
@@ -110,10 +110,15 @@ function(cub_build_compiler_targets)
   target_compile_options(cub.compiler_interface INTERFACE
     # If using CUDA w/ NVCC...
     # Display diagnostic numbers.
-    $<$<AND:$<COMPILE_LANGUAGE:CUDA>,$<CUDA_COMPILER_ID:NVIDIA>>:-Xcudafe=--display_error_number>
+    $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:-Xcudafe=--display_error_number>
     # Promote warnings.
-    $<$<AND:$<COMPILE_LANGUAGE:CUDA>,$<CUDA_COMPILER_ID:NVIDIA>>:-Xcudafe=--promote_warnings>
+    $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:-Xcudafe=--promote_warnings>
     # Don't complain about deprecated GPU targets.
-    $<$<AND:$<COMPILE_LANGUAGE:CUDA>,$<CUDA_COMPILER_ID:NVIDIA>>:-Wno-deprecated-gpu-targets>
+    $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:-Wno-deprecated-gpu-targets>
+    # Suppress deprecation warnings in nvcc < 11.5.
+    # TexRefInputIterator uses deprecated CUDART APIs, see NVIDIA/cub#191.
+    # After 11.5, we will suppress these in-code via pragma, but for older nvcc
+    # we have to use the big hammer:
+    $<$<AND:$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>,$<VERSION_LESS:$<CUDA_COMPILER_VERSION>,11.5>>:-Wno-deprecated-declarations>
```

Comment:
This is a good point. There are some definite advantages to just removing the deprecated class, but it's still a bit early IMO.

While CUB's version has increased since we deprecated this, we haven't shipped an updated version of CUB in the CTK yet. Depending on how future CUB/CTK release schedules line up, there might not be enough time for users to react to the deprecation if we remove the texture ref stuff now.

Let's give it a couple more releases before removing this entirely.